launcher/src/main/scripts/bin/launcher.py

```diff
@@ -189,6 +190,11 @@ def build_java_execution(options, daemon):
     if options.log_levels_set and not exists(options.log_levels):
         raise Exception('Log levels file is missing: %s' % options.log_levels)
 
+    try:
+        subprocess.check_call(['java -version'])
```

Comment:
I don't see how this works, since it's trying to run a binary named `java -version` (as `shell` defaults to `False`). It should be
```python
subprocess.check_call(['java', '-version'])

Reply:
Also, `check_call` doesn't handle the output, so it will be passed through. While showing the Java version at startup might be useful, I'd rather not make a behavioral change like that as part of this change.

We can handle it like this:
```python
subprocess.check_output(['java', '-version'], stderr=subprocess.STDOUT)

Reply:
Sorry that was dumb. I tested that the error was thrown when java wasn't installed but didn't make sure it wasn't thrown when it is. Will fix.

Reply:
Thanks. Updated and tested both cases on python 2 and 3.